src/Test.UnitTests.Sarif/Baseline2/MatchedResultsTests.cs

```diff
@@ -0,0 +1,291 @@
+                new FirstDetectionTimeTestCase
+                {
+                    Name = "New, provenance, no time information",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result
+                    {
+                        Provenance = new ResultProvenance()
+                    },
+                    CurrentRun = new Run(),
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = null
+                },
+                new FirstDetectionTimeTestCase
+                {
+                    Name = "New, provenance with time information",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result
+                    {
+                        Provenance = new ResultProvenance
+                        {
+                            FirstDetectionTimeUtc = firstDetectionTime
+                        }
+                    },
+                    CurrentRun = new Run(),
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = firstDetectionTime
+                },
+                new FirstDetectionTimeTestCase
+                {
+                    Name = "New, current run end time only",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result(),
+                    CurrentRun = new Run
+                    {
+                        Invocations = new List<Invocation>
+                        {
+                            new Invocation
+                            {
+                                EndTimeUtc = currentRunEndTime
+                            }
+                        }
+                    },
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = currentRunEndTime
+                },
+                new FirstDetectionTimeTestCase
+                {
+                    Name = "New, current run start time only",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result(),
+                    CurrentRun = new Run
+                    {
+                        Invocations = new List<Invocation>
+                        {
+                            new Invocation
+                            {
+                                StartTimeUtc = currentRunStartTime
+                            }
+                        }
+                    },
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = currentRunStartTime
+                },
+                new FirstDetectionTimeTestCase // TODO TEST CASE SHOWING CURRENT RUN WINS FOR NEW AND PREV RUN WINS FOR EXISTING
```

Comment:
Still something to do given comment?

Reply:
Good catch, I added the case.

---
In reply to: [372134533](https://github.com/microsoft/sarif-sdk/pull/1739#discussion_r372134533) [](ancestors = 372134533)